### PR TITLE
Fix Import Errors

### PR DESCRIPTION
**Pull request by s2t2**
## The Issue

I have been running into some Import Errors when trying to list or load the models from torch hub, for example:

```sh
import torch

pprint(torch.hub.list("unitaryai/detoxify"))

#> ImportError: cannot import name 'toxic_albert' from 'detoxify'
```

## The Fix

This PR adds some imports to the `detoxify/__init__.py` file, in hopes of fixing those import errors.

FYI - can install this version via:

```sh
pip install git+https://github.com/s2t2/detoxify.git@fix
```

### CODE REVIEW

**Comment by jamt9000** [comment]
Awesome, thanks!

It looks like running `pytest` in the repo root will fail when it tries to collect hubconf.py (which isn't a test) but if you run

```
pytest tests
```

tests pass for me in current master

File: tests/test_trainer.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def test_trainer():
 
     results = initialize_trainer(CONFIG)
     print(results)
-    assert results[0]["test_acc"] > 0.6
+    assert results[0]["test_acc"] > 0.3
```
Comment (jamt9000) [inline tests/test_trainer.py:65]: Was this giving a value much below 0.6 for you? It's giving 0.6992337703704834 deteministically when I run it on my mac, although it isn't really trained on much.

**Comment by jamt9000** [comment]
Since the tests will now cover the detoxify package this line in CI should be changed to `--source src,detoxify` so the coverage printed is correct

https://github.com/unitaryai/detoxify/blob/18fd29e97bf42fa1f741bbf9027c79e7d22c1900/.github/workflows/ci-testing.yml#L65

File: tests/test_trainer.py (hunk 1 above)
Comment (s2t2) [reply tests/test_trainer.py:65]: Weird. Yeah it was giving me like 37%. See screen shot. There was some additional output as well (pasted below). 

Screen Shot 2021-07-03 at 8 45 36 PM



```
tests/test_trainer.py:65: AssertionError
----------------------------------- Captured stdout call -----------------------------------
Epoch 0: 100%|██████████| 4/4 [00:34<00:00,  8.70s/it, loss=0.719, v_num=3]
Testing: 100%|██████████| 26/26 [00:25<00:00,  1.03it/s] 
--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{'test_acc': 0.3728836178779602, 'test_loss': 0.5584720969200134}
--------------------------------------------------------------------------------
[{'test_loss': 0.5584720969200134, 'test_acc': 0.3728836178779602}]
----------------------------------- Captured stderr call -----------------------------------
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertForSequenceClassification: ['cls.predictions.transform.dense.weight', 'cls.predictions.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.decoder.weight', 'cls.seq_relationship.bias', 'cls.predictions.transform.dense.bias', 'cls.seq_relationship.weight', 'cls.predictions.transform.LayerNorm.bias']
- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of BertForSequenceClassification were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
------------------------------------ Captured log call -------------------------------------
INFO     pytorch_lightning.utilities.seed:seed.py:65 Global seed set to 1234
INFO     pytorch_lightning.utilities.distributed:accelerator_connector.py:591 GPU available: False, used: False
INFO     pytorch_lightning.utilities.distributed:accelerator_connector.py:593 TPU available: False, using: 0 TPU cores
INFO     pytorch_lightning.core.lightning:lightning.py:1496 
  | Name  | Type                          | Params
--------------------------------------------------------
0 | model | BertForSequenceClassification | 109 M 
--------------------------------------------------------
109 M     Trainable params
0         Non-trainable params
109 M     Total params
437.947   Total estimated model params size (MB)
INFO     pytorch_lightning.utilities.seed:seed.py:65 Global seed set to 1234
```

Comment (jamt9000) [reply tests/test_trainer.py:65]: Strange... @laurahanu is a high value necessarily expected here?

Comment (s2t2) [reply tests/test_trainer.py]: FYI - I temporarily changed back to 0.6 to show the failing test on CI. You can see it here:

https://github.com/unitaryai/detoxify/pull/20/checks?check_run_id=2984564970#step:8:278

Now I'm changing it to 0.3, so tests will pass.

Comment (jamt9000) [reply tests/test_trainer.py:65]: Interesting, checked that it is also [failing with CI on master](https://github.com/unitaryai/detoxify/pull/23). I tried upgrading to `torch==1.9.0` `pytorch-lightning==1.3.8` on my laptop and it was still passing (0.699992835521698) but then also upgrading `transformers` (from 3.4.0 to 4.8.2) it fails (0.3728836178779602). Hopefully just random seed variability, but we should check the models are still working as expected with the newer versions of `transformers`.

Comment (laurahanu) [reply tests/test_trainer.py:65]: The 0.6 was chosen somewhat arbitrarily so it shouldn't matter that much if we change it to a lower value. I've checked and we still get the same values with the same input with the newest `torch==1.9.0` and `transformers==4.8.2` so the models should be working as expected.

image
image

Seems like the culprit here is [`transformers==4.6.0`](https://github.com/huggingface/transformers/releases/tag/v4.6.0) with this [specific PR](https://github.com/huggingface/transformers/pull/11471) where they introduce a fast initialisation of BERT. To get the same results as before we can just set `_fast_init=False` in:

```python
model = BertForSequenceClassification.from_pretrained("bert-base-cased", _fast_init=False)
```

**Review by jamt9000 — APPROVED** [review]
Ok, seems the 0.3 threshold should be fine, so LGTM!

**Comment by laurahanu** [comment]
Are we finw with the training not being reproducible though?

**Comment by jamt9000** [comment]
> Are we fine with the training not being reproducible though?

I think the training will still not be reproducible either way (eg. 0.6992337703704834 vs 0.699992835521698 above)

**Comment by laurahanu** [comment]
True, happy to carry on with 0.3.

**Review by laurahanu — APPROVED** [review]
Thanks for the PR @s2t2!

**Pull request merged**